assets/js/googlesitekit/widgets/util/get-widget-layout.test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Widgets layout utilities tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { getWidgetLayout } from './get-widget-layout';
+import { WIDGET_WIDTHS } from '../datastore/constants';
+
+describe( 'getWidgetLayout', () => {
+	const quarter = { width: WIDGET_WIDTHS.QUARTER };
+	const half = { width: WIDGET_WIDTHS.HALF };
+	const full = { width: WIDGET_WIDTHS.FULL };
+	const quarterOrHalf = { width: [ WIDGET_WIDTHS.QUARTER, WIDGET_WIDTHS.HALF ] };
+	const halfOrQuarter = { width: [ WIDGET_WIDTHS.HALF, WIDGET_WIDTHS.QUARTER ] };
+	const fullOrHalf = { width: [ WIDGET_WIDTHS.FULL, WIDGET_WIDTHS.HALF ] };
+
+	it( 'computes expected class names', () => {
+		const widgets = [
+			quarter,
+			quarter,
+			half,
+			half,
+			quarter,
+			full,
+		];
+		const expectedClassNames = [
+			[
+				'mdc-layout-grid__cell',
+				'mdc-layout-grid__cell--span-2-phone',
+				'mdc-layout-grid__cell--span-3-desktop',
+				'mdc-layout-grid__cell--span-4-tablet',
+			],
+			[
+				'mdc-layout-grid__cell',
+				'mdc-layout-grid__cell--span-2-phone',
+				'mdc-layout-grid__cell--span-3-desktop',
+				'mdc-layout-grid__cell--span-4-tablet',
+			],
+			[
+				'mdc-layout-grid__cell',
+				'mdc-layout-grid__cell--span-6-desktop',
+				'mdc-layout-grid__cell--span-8-tablet',
+			],
+			[
+				'mdc-layout-grid__cell',
+				'mdc-layout-grid__cell--span-8-desktop',
+				'mdc-layout-grid__cell--span-8-tablet',
+			],
```

Comment:
I'm starting to see what's going on but it is rather complicated. Would you please add comments as to why the various classes are expected for each?